migration/idempotent/007-tracing-tags.sql

```diff
@@ -497,8 +497,136 @@ AS $func$
     SELECT a.value #>> '{}'
     FROM _ps_trace.tag a
     WHERE a.key = _key -- partition elimination
-    AND a.id = (_tag_map ->> (SELECT id::text FROM _ps_trace.tag_key WHERE key = _key))::bigint
+    AND a.id = (_tag_map ->> (SELECT id::pg_catalog.text FROM _ps_trace.tag_key WHERE key = _key))::pg_catalog.bigint
     LIMIT 1
 $func$
 LANGUAGE SQL STABLE PARALLEL SAFE STRICT;
 GRANT EXECUTE ON FUNCTION ps_trace.val_text(ps_trace.tag_map, ps_trace.tag_k) TO prom_reader;
+
+-------------------------------------------------------------------------------
+-- Tag_map related functions and operators
+-------------------------------------------------------------------------------
+
+CREATE FUNCTION _ps_trace.tag_map_denormalize(_map ps_trace.tag_map)
+    RETURNS ps_trace.tag_map
+    LANGUAGE sql STABLE
+    PARALLEL SAFE AS
+/* NOTE: This function cannot be inlined since it's used in a scalar
+ * context and uses an aggregate. Yet we don't want to use `SET` clause
+ * for the search_path, thus it is important to make it completely
+ * search_path-agnostic.
+ */
+$fnc$
+    SELECT pg_catalog.jsonb_object_agg(t.key, t.value)
+        FROM pg_catalog.jsonb_each(_map) f(k,v)
+            JOIN _ps_trace.tag t ON f.v::pg_catalog.int8 OPERATOR(pg_catalog.=) t.id;
```

Comment:
just a note: this is fine now that we removed partitioning on tag, but if we reintroduce it, we'll want to be sure the implementation here invokes partition elimination